#!/usr/bin/env python
# SLAW OBJECT DETECTION

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t, str_t
#from local_planner_limits import add_generic_localplanner_params

gen = ParameterGenerator()


gen.add("save", bool_t, 0, "height of camera", False)
gen.add("cameraHeight", double_t, 0, "height of camera", 500, 0, 2000)
gen.add("pitch", double_t, 0, "pitch of camera", 50, -180,180 )
gen.add("yaw", double_t, 0, "pitch of camera", 0, -180, 180)
gen.add("imageWidth", int_t, 0, "height of camera", 640, 0, 4000)
gen.add("imageHeight", double_t, 0, "height of camera", 480, 0, 4000)
gen.add("ipmWidth", int_t, 0, "height of camera", 320, 0, 4000)
gen.add("ipmHeight", double_t, 0, "height of camera", 240, 0, 4000)
gen.add("ipmLeft", double_t, 0, "height of camera", 0, 0, 600)
gen.add("ipmRight", int_t, 0, "image width", 600, 0, 4000)
gen.add("ipmTop", int_t, 0, "image width", 220, 0, 4000)
gen.add("ipmBottom", int_t, 0, "image width", 420, 0, 4000)
gen.add("ipmInterpolation", bool_t, 0, "image width", True)

gen.add("alpha", double_t, 0, "height of camera", 0, 0, 360)
gen.add("beta", double_t, 0, "height of camera", 0, 0, 360)
gen.add("gamma", double_t, 0, "height of camera", 0, 0, 360)
gen.add("dx", double_t, 0, "height of camera", 0, 0, 360)
gen.add("dy", double_t, 0, "height of camera", 0, 0, 360)
gen.add("dz", double_t, 0, "height of camera", 0, 0, 360)
gen.add("f", double_t, 0, "height of camera", 0, 0, 360)


exit(gen.generate("slaw_object_detection", "ExtrinsicCalibr", "CamCalibrExtr"))
